src/napari_spatialdata/_model.py

```diff
@@ -78,7 +79,12 @@ def get_obs(
         """
         if name not in self.adata.obs.columns:
             raise KeyError(f"Key `{name}` not found in `adata.obs`.")
-        return self.adata.obs[name], self._format_key(name)
+        if name != self.instance_key:
+            adata_obs = self.adata.obs[[self.instance_key, name]]
+            adata_obs.set_index(self.instance_key, inplace=True)
+        else:
+            adata_obs = self.adata.obs
+        return adata_obs[name], self._format_key(name)
```

Comment:
I find this function a bit confusing. Here is why:
- In general, I would use the `get_values()` function from `SpatialData`. But here (at leasta for the time being) since we are using `AnnData` as a data model and since there is the subset operation that complicates things, let's keep it like this.

Anywhay:
- [ ] I find confusing that the behavior of the function is different depending on the value being passed being instance_key or not. Why does this happen? I woudl use two seprate functions.
- [ ] Also, the instance_key columns may be numeric (most of the time it is), while `AnnData` converts numerical columns to string. This may create problems, please consider this in this case.

Reply:
`get_obsm()` is actually not supported by `get_values()`, so that's another argument for keeping things like this for the time being.

Reply:
yeah the latter comment is exactly why I did not go for that approach. As for the different behaviour, in the `else` part, the user is basically wanting to plot the name of the `instance_key` column. If we would use the code in the `if` conditional this would result in a duplicate column which then results in a problem when setting the index.

Reply:
Btw regarding your second point, this does not get converted to string as we are not returning a full anndata object here. We are working with a subset of obs so it is not validated by anndata and thus not converted.